Sources/SwiftyRequest/RestRequest.swift

```diff
@@ -16,43 +16,56 @@
 
 import Foundation
 import CircuitBreaker
+import LoggerAPI
 
 /// Object containing everything needed to build HTTP requests and execute them
-public class RestRequest {
-
+public class RestRequest: NSObject  {
+    
+    // Check if there exists a self-signed certificate and whether it's a secure connection
+    private let isSecure: Bool
+    private let isSelfSigned: Bool
+    
     /// A default `URLSession` instance
-    private let session = URLSession(configuration: URLSessionConfiguration.default)
-
+    private var session: URLSession {
+        var session = URLSession(configuration: URLSessionConfiguration.default)
+        if isSecure && isSelfSigned {
+            let config = URLSessionConfiguration.default
+            config.requestCachePolicy = .reloadIgnoringLocalCacheData
+            session = URLSession(configuration: config, delegate: self, delegateQueue: .main)
+        }
+        return session
+    }
+    
     // The HTTP Request
     private var request: URLRequest
-
+    
     /// `CircuitBreaker` instance for this `RestRequest`
-    public var circuitBreaker: CircuitBreaker<(Data?, HTTPURLResponse?, Error?) -> Void, Void, String>?
```

Comment:
The first type declaration in for the CircuitBreaker is the invocation command argument. We want to pass `(Data?, HTTPURLResponse?, Error?) -> Void`, which can then be used to pass back the URLRequest result. 

The second type declaration can be removed (`Void`). It denoted the result type of the command execution. Since all commands are now asynchronous in CircuitBreaker, we removed the need to explicitly state it.

`CircuitBreaker<(Data?, HTTPURLResponse?, Error?) -> Void, String>